dspace-api/src/main/java/org/dspace/discovery/FullTextContentStreams.java

```diff
@@ -0,0 +1,241 @@
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.annotation.Nullable;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.sql.SQLException;
+import java.util.Enumeration;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.dspace.core.Utils.emptyIfNull;
+
+/**
+ * Construct a <code>ContentStream</code> from a <code>File</code>
+ */
+public class FullTextContentStreams extends ContentStreamBase
+{
+    private static final Logger log = Logger.getLogger(FullTextContentStreams.class);
+
+    public static final String FULLTEXT_BUNDLE = "TEXT";
+
+    protected final Context context;
+    protected List<FullTextBitstream> fullTextStreams;
+    protected List<FullTextBitstream> accessibleFullTextStreams;
+    protected BitstreamService bitstreamService;
+    protected AnonymousAccessChecker anonymousAccessChecker = new AnonymousAccessChecker();
+
+    public FullTextContentStreams(Context context, Item parentItem) throws SQLException {
+        this.context = context;
+        init(parentItem);
+    }
+
+    protected void init(Item parentItem) {
+        fullTextStreams = new LinkedList<>();
+        accessibleFullTextStreams = new LinkedList<>();
+ 
+        if(parentItem != null) {
+            sourceInfo = parentItem.getHandle();
+
+            //extracted full text is always extracted as plain text
+            contentType = "text/plain";
+
+            buildFullTextList(parentItem);
+        }
+    }
+
+    class AnonymousAccessChecker {
+        public boolean check(Bitstream bit) {
+            try {
+                return AuthorizeServiceFactory.getInstance().getAuthorizeService().authorizeActionBoolean(new Context(), bit, Constants.READ);
+            } catch (Exception e) {
+                log.error("Error checking bitstream access", e);
+                return false;
+            }
+        }
+    }
+    
+    private void buildFullTextList(Item parentItem) {
+        // now get full text of any bitstreams in the TEXT bundle
+        // trundle through the bundles
+        List<Bundle> myBundles = parentItem.getBundles();
+
+        for (Bundle myBundle : emptyIfNull(myBundles)) {
+            if (StringUtils.equals(FULLTEXT_BUNDLE, myBundle.getName())) {
+                // a-ha! grab the text out of the bitstreams
+                List<Bitstream> bitstreams = myBundle.getBitstreams();
+
+                for (Bitstream fulltextBitstream : emptyIfNull(bitstreams)) {
+                
+                    FullTextBitstream ftBitstream = new FullTextBitstream(sourceInfo, fulltextBitstream);
+                    fullTextStreams.add(ftBitstream);
+                    boolean isAccessible = anonymousAccessChecker.check(fulltextBitstream);
+                        
+                    if (isAccessible) {
+                        accessibleFullTextStreams.add(ftBitstream);
+                    }    
```

Comment:
As mentioned in the comments of https://github.com/DSpace/DSpace/pull/1595, I don't think this is enough to support policies with a start or end date (embargoes) without doing a full Discovery reindex.